samples/msal-node-samples/nextjs-rsc-ts/src/components/WelcomeName.tsx

```diff
@@ -0,0 +1,16 @@
+import { authProvider } from "~/services/auth";
+import { Typography } from "./mui";
+
+export default async function WelcomeName() {
+  const account = await authProvider.getAccount();
```

Comment:
By default, all components in Next.js that are used in the App Router are considered server components. Only components marked with the `"use client";` directive are rendered on the server then hydrated on the client.

Source: https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive

The use of the `"use server";` directive further up I believe is part of the server actions RFC that is currently experimental in Next.js/React and imho should probably be avoided using in docs until it is stable. Though it could be split into a separate example so when they are stable the example is already there.

Reply:
It will only work in server components. As stated all components in App Router are server by default.

I will however poison the authProvider modules with `import "server-only"` to avoid them accidentally being imported in client components.

https://nextjs.org/docs/getting-started/react-essentials#keeping-server-only-code-out-of-client-components-poisoning